src/script/api/script_list.cpp

```diff
@@ -937,25 +938,30 @@ SQInteger ScriptList::Valuate(HSQUIRRELVM vm)
 	ScriptObject::DisableDoCommandScope disabler{};
 
 	/* Limit the total number of ops that can be consumed by a valuate operation */
-	SQOpsLimiter limiter(vm, MAX_VALUATE_OPS, "valuator function");
+	SQOpsLimiter limiter(vm, ScriptController::GetOpsTillSuspend(), "valuator function");
```

Comment:
```suggestion
	SQOpsLimiter limiter(vm, _settings_game.script.script_max_opcode_till_suspend, "valuator function");
```
This solves the issue of item repeat. I have tried different ways to break it and so far I couldn't. I tried a list with 10 million items, and I tried a valuator function using as many ops as it could without going into excessive cpu error. It still iterates all the items without repeating.

Reply:
But doing that makes the tick use more (double in worst cases, when `Valuate()` happens at the end of a tick) than max opcode.